main.cc

```diff
@@ -25,17 +25,25 @@ static bool reliable_write(int fd, const char *buffer, int len) {
 }
 
 int main(int argc, char *argv[]) {
-    if (argc != 2) {
-        fprintf(stderr, "usage: %s <gcode-file>\n", argv[0]);
+    if (argc < 2) {
+        fprintf(stderr, "usage: %s <gcode-file> [connection string]\n", argv[0]);
+        fprintf(stderr, "Example: %s file.gcode /dev/ttyACM0,b115200\n", argv[0]);
+
         return 1;
     }
     const char *filename = argv[1];
+
+    std::string connection_string="/dev/ttyACM0,b115200";
```

Comment:
Given that the alternative string we get from argv[2] which alive when we call OpenMachineConnection(), we don't need to copy it into a std::string; so we can just assign it to a `const char *`. That way, we also don't have to do the c_str() conversion below in the OpenMachineConnection() and fprintf() call.

```
const char *connection_string = "/dev/ttyACM0,b115200";
```
Also please add a space between `if` and the open parenthesis to match the rest of the style in the file. Also, spaces around the assignment operator.